### PR TITLE
add isDartCorePrint extension method

### PR DESCRIPTION
**Pull request by pq**
Follow-up from: https://github.com/dart-lang/linter/pull/3379#discussion_r867170225

/cc @scheglov @a14n

### CODE REVIEW

File: lib/src/ast.dart [hunk 1]
```diff
@@ -434,3 +434,11 @@ class _ElementVisitorAdapter extends GeneralizingElementVisitor {
     }
   }
 }
+
+extension ElementExtension on Element? {
+  bool get isDartCorePrint {
+    var element = this;
```
Comment (scheglov) [inline lib/src/ast.dart:440]: In analyzer we prefer making this `final self = this;`.

File: lib/src/ast.dart [hunk 2]
```diff
@@ -434,3 +434,11 @@ class _ElementVisitorAdapter extends GeneralizingElementVisitor {
...
+    if (element is! FunctionElement) return false;
```
Comment (scheglov) [inline lib/src/ast.dart:441]: Even less important, but just note that `element is FunctionElement` would fit well into the following `&&` expression. Although there will be multiple lines :-)

**Review by scheglov — APPROVED** [review]

File: lib/src/ast.dart (hunk 2 above)
Comment (pq) [reply lib/src/ast.dart:441]: I do actually like that better.  Updated.  Thanks!

File: lib/src/ast.dart (hunk 1 above)
Comment (pq) [reply lib/src/ast.dart]: Ah, nice convention.

**Pull request merged**